src/Core/Cache/InMemoryCacheFactory.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace SilverStripe\Core\Cache;
+
+use Psr\Cache\CacheItemPoolInterface;
+
+/**
+ * Interface to be implemented by all cache factories that instantiate in-memory symfony cache adapters.
+ * This allows the cache adapters to be used in conjunction with filesystem cache.
+ */
+interface InMemoryCacheFactory extends CacheFactory
+{
+    /**
+     * Create a PSR6-compliant cache adapter
+     */
+    public function createPsr6(string $service, array $params = []): CacheItemPoolInterface;
```

Comment:
PSR6 specifically required so it can be bundled in a `ChainAdapter` along-side the filesystem cache adapter.
This is in contrast to the `create()` method which must be PSR16 for direct use.